.github/workflows/appimage.yml

```diff
@@ -26,12 +26,25 @@ jobs:
     timeout-minutes: 30
 
     steps:
-    - name: Check out code in viam-cartographer directory
+    - name: Check out main code
+      if: github.event_name == 'workflow_dispatch' || github.event_name == 'push'
       uses: actions/checkout@v3
       with:
         submodules: recursive
         path: viam-cartographer
+      
+    - name: Check out PR branch code
+      if: github.event_name == 'pull_request_target'
+      uses: actions/checkout@v3
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
+        submodules: recursive
+        path: viam-cartographer
 
+    - name: make clean
+      run: |
```

Comment:
I'd add a leading line here
`chown -R testbot:testbot .` to change the ownership to the testbot user and prevent potential issues.

Reply:
Why `chown -R testbot:testbot .` instead of `chown -R testbot .`?